main.js

```diff
@@ -177,19 +226,37 @@ define(function (require, exports, module) {
 
     function onCheckedStateChange() {
         _preferences.set("checked", Boolean(_command.getChecked()));
+        _preferences.set("colors", _preferences.get("colors"));
         updateEditors();
     }
-    
+
     function onActiveEditorChange(e, editor) {
         updateEditors(editor);
     }
 
+    function updateColors(e, data) {
+        if (data.ids.indexOf("theme") > -1 || data.ids.indexOf("colors") > -1) {
+            _applyColors();
+        }
+    }
+
     
     // --- Loaders and Unloaders ---
 
     function loadPreferences() {
         _preferences = PreferencesManager.getExtensionPrefs(preferencesId);
-        _preferences.definePreference("checked", "boolean", defaultPreferences["checked"]);
+        _preferences.definePreference("checked", "boolean", defaultPreferences.checked);
+        _preferences.definePreference("colors", "Object", defaultPreferences.colors);
+    }
+  
+    function loadPrefListeners() {
+        _preferences.on("change", updateColors);
+        PreferencesManager.getExtensionPrefs("themes").on("change", updateColors);
+    }
+  
+    function unloadPrefListeners() {
+        _preferences.off("change");
+        PreferencesManager.getExtensionPrefs("themes").off("change");
```

Comment:
I am wondering if this will turn off all registered listeners...  Would it be more correct to unregister only the method you registered by passing the callback?